pilot/pkg/model/push_context.go

```diff
@@ -1082,8 +1082,7 @@ func (ps *PushContext) getSidecarScope(proxy *Proxy, workloadLabels labels.Insta
 		}
 
 		// We need to compute this namespace
-		computed := DefaultSidecarScopeForNamespace(ps, proxy.ConfigNamespace)
-		ps.sidecarIndex.defaultSidecarsByNamespace[proxy.ConfigNamespace] = computed
```

Comment:
why delete this line ?

Reply:
It is intended, because now the sidecarscope initiated from gateway is not complete. 

To improve a little bit, we can add a new cache for gateway only

Reply:
That's a good input, are you interested in taking it

Reply:
https://github.com/istio/istio/pull/49899